### PR TITLE
Change string matching method to avoid encoding error

### PR DESCRIPTION
**Pull request by iriberri**
The previous string matching method that was used would not match the regexp if not all code points are valid.

This was provoking import errors in Shapefiles with names that contained special characters. Fixes https://github.com/CartoDB/cartodb/issues/6276.

Quick example:

````ruby
irb(main):001:0> a = "\xFF\xFF\xFFhello"
=> "\xFF\xFF\xFFhello"
irb(main):002:0> a.encoding
=> #<Encoding:UTF-8>
irb(main):003:0> a.valid_encoding?
=> false
irb(main):004:0> a.include? "hello"
=> true
irb(main):005:0> a =~ /hello/
ArgumentError: invalid byte sequence in UTF-8
````

### CODE REVIEW

**Comment by iriberri** [comment]
cc @rafatower

**Comment by iriberri** [comment]
I fixed one text matching, but there are quite a few more in the same file that were working in local but are raising errors in the staging environment.

**Comment by iriberri** [comment]
Ops. It seems that removing the error message comparison in the unp output is breaking some tests.

File: services/importer/spec/unit/unp_spec.rb [hunk 1]
```diff
@@ -231,7 +231,7 @@
 
   describe '#unp_failure?'  do
     it 'returns true if unp cannot read the file' do
-      Unp.new.unp_failure?('Cannot read', 0).should eq true
```
Comment (rafatower) [inline services/importer/spec/unit/unp_spec.rb]: this was the wrong assumption in the past.

**Comment by iriberri** [comment]
retest this please

**Comment by rafatower** [comment]
there are a number of tests broken, probably by me :(

let me know if you need help fixing it.

**Comment by iriberri** [comment]
I have been working in this branch in a new attempt to run everything from scratch: https://github.com/CartoDB/cartodb/compare/6276-unp_filenames_encoding

Haven't PR'd yet, as I managed to "get it working" last day at 10PM and still wanted to think about it because the approach I followed is quite aggressive.

**Comment by iriberri** [comment]
Closing without merging. We can handle the error conveniently (instead of raising an unknown error).

**Pull request closed without merging**